dependencies.yaml

```diff
@@ -0,0 +1,107 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+ files:
+   all:
+     output: conda
+     matrix:
+       cuda: ["11.5"]
+       arch: [x86_64]
+     includes:
+       - common_build
+       - cpp_build
+       - cudatoolkit
+       - doc
+       - python_build
+ channels:
+   - rapidsai
+   - rapidsai-nightly
+   - dask/label/dev
+   - conda-forge
+   - nvidia
+ dependencies:
+   cudatoolkit:
+     specific:
+       - output_types: conda
+         matrices:
+           - matrix:
+               cuda: "11.5"
+             packages:
+               - cudatoolkit=11.5
+           - matrix:
+               cuda: "11.4"
+             packages:
+               - cudatoolkit=11.4
+           - matrix:
+               cuda: "11.2"
+             packages:
+               - cudatoolkit=11.2
+           - matrix:
+               cuda: "11.0"
+             packages:
+               - cudatoolkit=11.0
+   common_build:
+     common:
+       - output_types: [conda]
+         packages:
+           - cmake>=3.23.1,!=3.25.0
+       - output_types: conda
+         packages:
+           - ninja
+           - pip
+   cpp_build:
+     common:
+       - output_types: conda
+         packages:
+           - nvcc_linux-64=11.5
+           - c-compiler
+           - cxx-compiler
+           - librmm=23.02.*
+           - libraft-headers=23.02.*
+           - libraft-distance=23.02.*
+           - libcugraphops=23.02.*
+           - clang=11.1.0
+           - clang-tools=11.1.0
+           - boost
+           - gtest=1.10.0
+           - gmock=1.10.0
+   python_build:
+     common:
+       - output_types: conda
+         packages:
+           - rmm=23.02.*
+           - cudf=23.02.*
+           - libcudf=23.02.*
+           - raft-dask=23.02.*
+           - pylibraft=23.02.*
+           - cuda-python>=11.7.1,<12.0
+           - dask-cuda=23.02.*
+           - dask-cudf=23.02.*
+           - dask>=2022.11.1
+           - distributed>=2022.11.1
+           - nccl>=2.9.9
+           - ucx-py=0.30.*
+           - ucx-proc=*=gpu
+           - scipy
+           - networkx>=2.5.1
+           - scikit-build>=0.13.1
+           - python>=3.8,<3.10
```

Comment:
Let's use a separate dependency list for Python versions. See cudf example: https://github.com/rapidsai/cudf/blob/2048829ec66af1ceb7a9801702417d12ef13c77a/dependencies.yaml#L190-L204